packages/website/docs/state.md

```diff
@@ -67,51 +47,75 @@ const autocomplete = createAutocomplete({
 });
 ```
 
+You can also manually update the state using setters. It's useful to implement custom features on top of autocomplete.
+
+In the following example, when detecting a potential filter in the query, you're manually adding it to the list of active filters using [Context](context) and removing it from the query.
+
+```js
+const categories = ['Tops', 'Bottoms', 'Hats', 'Bags', 'Footwear'];
+
+const autocomplete = createAutocomplete({
+  // ...
+  onStateChange({ state, setQuery, setContext }) {
+    const { query, context } = state;
+    const match = categories.find((category) => query.endsWith(category));
+
+    if (match) {
+      const filters = [...context.filters, match];
+      const position = query.lastIndexOf(match);
+
+      setContext({ filters });
+      setQuery(query.substr(0, position));
+    }
+  },
+});
+```
+
 ## State
 
 ### `activeItemId`
 
 > `number | null` | defaults to `null`
 
-The highlighted item index.
+The highlighted item's index.
 
 ### `query`
 
 > `string` | defaults to `""`
 
-The query of the input.
+The value of the search input.
 
 ### `completion`
 
 > `string | null` | defaults to `null`
 
-The completion of the input.
+The completion of the query.
 
 ### `isOpen`
 
 > `boolean` | defaults to `false`
 
-Whether the panel is opened.
+Whether the panel is open or not.
 
 ### `collections`
 
 > `Collection[]` | defaults to `[]`
 
-The collections of the experience.
+The collections of items.
 
 ### `status`
 
 > `'idle' | 'loading' | 'stalled' | 'error'` | defaults to `idle`
 
-The status of the experience.
+The autocomplete's status.
 
 ### `context`
 
-> `object` | defaults to `{}`
+> `{ [key: string]: unknown }` | defaults to `{}`
```

Comment:
```suggestion
> `Record<string, unknown>` | defaults to `{}`
```

Or if we introduce custom types in the doc, we can directly use the `AutocompleteContext` type?

Reply:
Agreed.
```suggestion
> `AutocompleteContext` | defaults to `{}`
```